src/rebar3_sbom_cpe.erl

```diff
@@ -0,0 +1,82 @@
+-module(rebar3_sbom_cpe).
+
+-export([hex/3]).
+
+% Includes
+-include("rebar3_sbom.hrl").
+
+%--- Macros --------------------------------------------------------------------
+-define(CPE_PREFIX, <<"cpe:", ?CPE_VERSION/binary>>).
+
+% The CPE specs define 3 classes of parts:
+% - a: application
+% - o: operating system
+% - h: hardware
+% We only support application components for now.
+-define(CPE_PART_APPLICATION, <<"a">>).
+
+%--- Types ---------------------------------------------------------------------
+
+-type cpe_opts() :: #{update => bitstring(),
+                      edition => bitstring(),
+                      language => bitstring(),
+                      target_sw => bitstring(),
+                      target_hw => bitstring()}.
+
+%--- API -----------------------------------------------------------------------
+
+-spec hex(Name, Version, Url) -> CPE when
+    Name :: bitstring(),
+    Version :: bitstring(),
+    Url :: bitstring() | undefined,
+    CPE :: bitstring().
+hex(<<"hex_core">>, Version, _) ->
+    <<?CPE_PREFIX/binary, ":", ?CPE_PART_APPLICATION/binary,
+      ":hex:hex_core:", Version/bitstring, ":*:*:*:*:*:*:*">>;
```

Comment:
I ended up counting the `*` here in every line for review. Maybe using a macro for the suffix would be good.